packages/editor/src/components/post-text-editor/index.js

```diff
@@ -7,32 +7,37 @@ import Textarea from 'react-autosize-textarea';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Component } from '@wordpress/element';
+import { useState } from '@wordpress/element';
 import { parse } from '@wordpress/blocks';
-import { withSelect, withDispatch } from '@wordpress/data';
-import { withInstanceId, compose } from '@wordpress/compose';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useInstanceId } from '@wordpress/compose';
 import { VisuallyHidden } from '@wordpress/components';
 
-export class PostTextEditor extends Component {
-	constructor() {
-		super( ...arguments );
+export default function PostTextEditor() {
+	const val = useSelect(
+		( select ) => select( 'core/editor' ).getEditedPostContent(),
+		[]
+	);
 
-		this.edit = this.edit.bind( this );
-		this.stopEditing = this.stopEditing.bind( this );
+	const { editPost, resetEditorBlocks } = useDispatch( 'core/editor' );
 
-		this.state = {};
+	const [ value, setValue ] = useState( val );
+	const [ isDirty, setIsDirty ] = useState( false );
+	const instanceId = useInstanceId( PostTextEditor );
+
+	if ( ! isDirty && value !== val ) {
+		setValue( val );
 	}
 
-	static getDerivedStateFromProps( props, state ) {
-		if ( state.isDirty ) {
-			return null;
-		}
+	const onChange = ( content ) => {
```

Comment:
This is used only in `edit` function which in turn is used at rendering at `onChange` event. You could add this line:
`editPost( { content } )` in `edit` function and rename it `onChange`. It was like this before due to the composition.

The result would be like:
```
onChange={ onChange }
```